src/createServer.js

```diff
@@ -4,0 +1,78 @@
+const http = require('http');
+const { detectCase } = require('./convertToCase/detectCase');
+const { convertToCase } = require('./convertToCase/convertToCase');
+
+const allCases = ['SNAKE', 'KEBAB', 'UPPER', 'CAMEL', 'PASCAL'];
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    const errors = [];
+
+    try {
+      const [path, queryString] = req.url.split('?');
+      const params = new URLSearchParams(queryString);
+
+      const targetCase = params.get('toCase');
+      const originalText = path.slice(1);
+      const originalCase = detectCase(originalText);
+
+      if (!originalText) {
+        errors.push({
+          message:
+            // eslint-disable-next-line max-len
+            'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+        });
+      }
```

Comment:
It's good that you're validating the user's input and providing helpful error messages. However, the error handling could be more efficient. Consider creating a separate function for input validation. This will make your code cleaner and easier to read.